doc/reference/networking/can_api.rst

```diff
@@ -0,0 +1,254 @@
+is also used in home and industrial automation and other products.
+
+A so-called CAN transceiver connects the two bus lines, CAN High (CAN H) and
+CAN Low (CAN L), from the CAN network to the CAN controller.
+The send wire from the controller to the transceiver is called CAN TX, and the
+receive wire is called CAN RX. These wires use the logic levels whereas the bus
+level is interpreted differential between CAN H and CAN L.
+Both lines, CAN H and CAN L, at the same level, is called the recessive state.
+This state is seen as a logical one, which is also the idle state.
+The other state is called the dominant state and is interpreted as a logical
+zero value. To write a dominant bit to the bus, open-drain transistors tight
+CAN H to Vdd and CAN L to ground. The first and last node use a 120 ohm resistor
+between CAN H and CAN L to terminate the bus.
+The termination avoids reflections on the bus and keeps CAN L and CAN H at the
+same level in idle state.
+The dominant state always overrides the recessive state.
+This structure is called a wired-AND.
+
+CAN controllers can only initialize when the bus is in the idle (recessive)
+state for a specific amount of time. Therefore you have to make sure that CAN RX
+is high, at least for a short time. This is also necessary for loopback mode.
+
+CAN uses Identifiers to identify the frame instead of addresses to identify a
+node. This identifier can either have 11-bit width (Standard or Basic Frame) or
+29-bit in case of an Extended Frame. The zephyr API supports both Standard and
+Extended Identifiers concurrently. A CAN frame starts with a dominant
+Start Of Frame bit. After that, the identifiers follow. This phase is called the
+arbitration phase. During the arbitration phase, write collisions are allowed.
+They resolve by the fact that dominant bits override recessive bits.
+Nodes that get overridden abort the transmission and retry when the bus is in
+the idle state again. It results in the fact that lower number identifiers have
+priority over higher number identifiers.
+
+Filters are used to whitelist identifiers that are of interest for the specific
+node. An identifier that doesn't match any filter is ignored.
+Filters can either match exactly or a specified part of the identifier.
+This method is called masking.
+As an example, a mask with 11 bits set for standard or 29 bits set for extended
+identifiers must match perfectly.
+Bits that are set to zero in the mask don't care for the matching.
+The filters are implemented in hardware for the most CAN controllers.
+The hardware therefor limits the number of filters.
+Additionally, the number of filters is limited in Kconfig, to save memory.
+
+
+`CAN Wikipedia article <https://en.wikipedia.org/wiki/CAN_bus>`_.
+
+Zephyr supports following CAN features:
+
+* Standard and Extended Identifers
+* Filters with Masking
+* Loopback and Silent mode
+* Remote Request
+
+Sending
+*******
+
```

Comment:
Should we also have some information about SocketCAN, at least mention that we support that and pointing a link to sample app.

Reply:
Sure. I'll add a link to SocketCAN in the overview.